### DISCUSSION TITLE
Custom CRT call not working in OFFLine mode

### DISCUSSION THREAD

**Discussion by DanyallAslam** [microsoft/Dynamics365Commerce.Solutions#46] [Q&A]
We have wrote custom CRT extension and it is working fine when we are in ONline mode but when we switch to Offline mode it is throwing error 
crt://offline/GasPumpManager/methodName?Cart=SCRUBBED&ConStatus=SCRUBBED&CustCNIC=SCRUBBED&$locale=SCRUBBED. Exception: System.InvalidOperationException: Can't find manager (GasPumpManager) with method (methodName) with parameter count (3)

**Comment by LavanyaNiki** [comment]
@DanyallAslam: 
We are using 9.47 version commerce SDK and when we switch to Offline mode all extensions are throwing an error. 

Ex: Exception: System.InvalidOperationException: Can't find manager (RetailParametersEntityManager) with method (methodName) with parameter count (3)

Retail proxy extension manager method (managerName: RetailParametersEntityManager, methodName: GetRetailParameters, parameterCount: 2) is not found in any proxy extension assembly under *.Adapters namespace. 

As per below blog Retail Server extension built using this the Microsoft.Dynamics.Commerce.Runtime.Hosting.Contracts API can be used in and offline implementation, no need to generate separate C# proxy library.

https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/typescript-proxy-retail-pos

If you already found the solution for this issue, please help me here.
